### PR TITLE
Add ConfirmationView to forms-system

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- Add `components/ConfirmationView` to `forms-system`
- This was prototyped in `simple-forms` directory, and now we are migrating it to `forms-system` for all teams to use.
> The forms-system `component/ConfirmationView` contains the standard view elements necessary for your application's `container/ConfirmationPage`. Can be used as a single component, or by using subcomponent pattern `ConfirmationView.ChildElements` pattern to omit/add/replace individual components.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team-forms#1750

## Testing done

- Browser and unit testing.
- Tested on forms 21-10210, 20-10206, mock-simple-form-patterns

## Screenshots

image
image
image

## What areas of the site does it impact?

Allows users to use this in their `ConfirmationPage` container.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/forms-system/src/js/components/ConfirmationView/ChapterSectionCollection.jsx [hunk 1]
```diff
@@ -0,0 +1,329 @@
+import React from 'react';
```
Comment (rhasselle-oddball) [inline src/platform/forms-system/src/js/components/ConfirmationView/ChapterSectionCollection.jsx:1]: This file is migrated from https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/simple-forms/shared/components/confirmationPageViewHelpers.jsx. Minimal changes here, you can compare the signature on line 264.

File: src/platform/forms-system/src/js/components/ConfirmationView/ConfirmationView.jsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import React, { useState } from 'react';
```
Comment (rhasselle-oddball) [inline src/platform/forms-system/src/js/components/ConfirmationView/ConfirmationView.jsx:1]: This file migrated from 
https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/simple-forms/shared/components/ConfirmationPageView.v2.jsx. There are major changes here, including breaking out all components into ./components file, and using a subcomponent pattern with useContext

**Review by rhasselle-oddball — COMMENTED** [review]
dev notes

File: src/platform/forms-system/src/js/components/ConfirmationView/README.md
Comment (pennja) [inline src/platform/forms-system/src/js/components/ConfirmationView/README.md:1]: Loving all the documentation! 😍

**Review by pennja — APPROVED** [review]
From what I can tell, this looks good! It's a huge PR and I know a lot of it is redundant code which already existed in some form or another but I'd be much happier if we could get another review from someone else! Regardless, approved!

**Review by ndsprinkle — APPROVED** [review]
I think everything looks great. I think the modular approach to the confirmation page is interesting. As always, I'm sure we'll want to encourage people use the full suite of the confirmation page since that's our attempt to standardize it, but there will no doubt be some circumstance that someone wants to remove some part or pass in their own content.

**Pull request merged**